### PR TITLE
AUT-2701: Improve management of redis connection resources in lambdas.

### PR DESCRIPTION
**Pull request by andrew-moores**
The lambdas create a lot of new redis connections each time they are invoked.  These conections are not closed explicitly.  We can improve the management of these connections by moving their creation to the lambda runtime init phase.

## Checklist

- [ ] Impact on orch and auth mutual dependencies has been checked.

- [ ] A UCD review has been performed.

### CODE REVIEW

**Review by BeckaL — APPROVED** [review]
Test approval - not merging

**Review by BeckaL — DISMISSED** [review]
Another test approval - not merging

**Comment by andrew-moores** [comment]
LGTM

**Review by pskushwaha1 — APPROVED** [review]
Approved as already reviewed by Andrew and Becka

File: frontend-api/src/main/java/uk/gov/di/authentication/frontendapi/lambda/CheckReAuthUserHandler.java [hunk 1]
```diff
@@ -30,11 +30,29 @@
...
-    private final CodeStorageService codeStorageService;
+    static {
+        if (System.getProperty("TEST") == null) {
+            redis = new RedisConnectionService(ConfigurationService.getInstance());
+        }
+    }
+
+    // FUNCTION INIT
+    public CheckReAuthUserHandler() {
+        this(ConfigurationService.getInstance(), redis);
```
Comment (andyloughran) [inline frontend-api/src/main/java/uk/gov/di/authentication/frontendapi/lambda/CheckReAuthUserHandler.java:45]: If the connection breaks; is there anything here that will re-connect - or will the lambda simply have a broken connection?

Comment (andrew-moores) [reply frontend-api/src/main/java/uk/gov/di/authentication/frontendapi/lambda/CheckReAuthUserHandler.java:45]: At the moment the first call that uses the broken connection will fail and cause an RTE to be thrown which, in turn, will cause the lambda to restart.  If we see improvement from this change then the next step will be to add in better handling of errors on the redis client including broken connections.

**Pull request merged**